### PR TITLE
feat(match): add 'me'-endpoint for matches

### PR DESCRIPTION
**Pull request by LarsSelbekk**

### CODE REVIEW

**Comment by LarsSelbekk** [comment]
[Skjule andre sine matches i APIet](https://trello.com/c/mOvznm3b/662-skjule-andre-sine-matches-i-apiet)

**Comment by LarsSelbekk** [comment]
I'm aware there's a TODO in there to move some types to bl-model, but I really can't be bothered to bump bl-model right now. If you want to, that's great.

**Review by AdrianAndersen — CHANGES_REQUESTED** [review]

File: src/collections/match/operations/match-getall-me.operation.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      ? matchStorage
+      : new BlDocumentStorage(BlCollectionName.Matches, matchSchema);
+  }
+
+  async run(blApiRequest: BlApiRequest): Promise<BlapiResponse> {
+    const meQuery = new SEDbQuery();
+    meQuery.stringFilters = [
+      {
+        fieldName: "blid",
+        value: blApiRequest.user.id,
```
Comment (AdrianAndersen) [inline src/collections/match/operations/match-getall-me.operation.ts:54]: Why not use the primary key for userDetail?

```suggestion
        fieldName: "_id",
        value: blApiRequest.user.id,
```

Comment (LarsSelbekk) [reply src/collections/match/operations/match-getall-me.operation.ts:42]: @AdrianAndersen

File: src/collections/match/operations/match-getall-me.operation.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+// TODO: Move these two to bl-model
+interface RelevantDetails {
+  name: string;
+}
+type MatchWithDetails =
+  | StandMatch
+  | (UserMatch & {
+      senderDetails: RelevantDetails;
+      receiverDetails: RelevantDetails;
+    });
```
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/collections/match/operations/match-getall-me.operation.ts]: Done

File: src/collections/match/operations/match-getall-me.operation.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    this.userStorage = userStorage
+      ? userStorage
+      : new BlDocumentStorage(BlCollectionName.Users, UserSchema);
+    this.userDetailStorage = userDetailStorage
+      ? userDetailStorage
+      : new BlDocumentStorage(BlCollectionName.UserDetails, userDetailSchema);
+    this.matchStorage = matchStorage
+      ? matchStorage
+      : new BlDocumentStorage(BlCollectionName.Matches, matchSchema);
+  }
```
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/collections/match/operations/match-getall-me.operation.ts]: Done

File: src/collections/match/operations/match-getall-me.operation.ts (hunk 1 above)
Comment (AdrianAndersen) [reply src/collections/match/operations/match-getall-me.operation.ts:41]: It certainly does include the userDetail primary key. 

However, it seems like the API `handleRequest` (higher order method) did not forward it to `blApiRequest` for some reason. That is stupid, so I changed it.😛

**Review by AdrianAndersen — APPROVED** [review]

File: src/collections/match/operations/match-getall-me.operation.ts (hunk 1 above)
Comment (LarsSelbekk) [reply src/collections/match/operations/match-getall-me.operation.ts]: Ah ok, I only looked at the actual properties in the debugger

**Pull request merged**